enhancements/storage/csi-secrets-store.md

```diff
@@ -0,0 +1,253 @@
+
+1. Implement a new secrets-store-csi-driver-operator using [CSI driver installation functions of library-go](https://github.com/openshift/library-go/tree/master/pkg/operator/csi/csicontrollerset). This is well tested and used by a dozen other CSI driver operators already.
+2. secrets-store-csi-driver-operator watches `ClusterCSIDriver` CR named `secrets-store.csi.k8s.io` and deploys the CSIDriver, RBAC roles, and node DaemonSet for the driver.
+3. Ship the operator + CSI driver through ART pipeline, same as other CSI drivers.
+4. Admin installs operator via OLM, then chooses a third-party provider plugin to install.
+5. User can deploy pods with external secrets mounted as inline ephemeral volumes.
+
+### Workflow Description
+
+#### Installing CSI Driver
+
+1. Cluster admin installs CSI driver operator via OLM.
+2. Cluster admin creates ClusterCSIDriver instance for the driver (`secrets-store.csi.k8s.io`).
+3. Cluster admin installs a third-party provider plugin for their chosen secret store.
+
+See also: [upstream CSI driver installation docs](https://secrets-store-csi-driver.sigs.k8s.io/getting-started/installation.html)
+
+#### Mounting External Secrets to a Pod
+
+1. Pod is created and scheduled to a node.
+2. Kubelet issues request to CSI driver to mount the volume.
+3. CSI driver sends request for the secret to provider plugin via a Unix domain socket.
+4. Provider retrieves the secret from the external secret store and returns it to the CSI driver.
+5. CSI driver creates an ephemeral volume, writes the secret to it, and returns the mount path to kubelet.
+6. Kubelet reports the volume is mounted and pod is running.
+
+#### Uninstalling CSI Driver
+
+1. Cluster admin stops all application pods that use the `secrets-store.csi.k8s.io` provider.
+2. Cluster admin deletes the `secrets-store.csi.k8s.io` ClusterCSIDriver object.
+3. The operator removes the CSI driver and associated manifests. Cluster admin verifies the CSI driver pods are no longer running.
+4. Cluster admin uninstalls the Secrets Store CSI driver operator via OLM. This should be done only after the CSI driver itself was already removed. Removing the operator does not automatically remove the driver.
+
+### API Extensions
+
+The only planned OpenShift API change is to add [secrets-store.csi.k8s.io](https://github.com/openshift/api/blob/bdd8865676216305be6065ade196f36752dfd015/operator/v1/types_csi_cluster_driver.go#L86) to the list of supported providers in the ClusterCSIDriver object.
+
+Once the operator is installed and ClusterCSIDriver object is created, the operator must apply two new CRD's required by the provider plugins:
+[secretproviderclasses](https://github.com/openshift/secrets-store-csi-driver/blob/main/deploy/secrets-store.csi.x-k8s.io_secretproviderclasses.yaml) and
+[secretproviderclasspodstatuses](https://github.com/openshift/secrets-store-csi-driver/blob/main/deploy/secrets-store.csi.x-k8s.io_secretproviderclasspodstatuses.yaml).
```

Comment:
We should mention (somewehere else?) that the operator installs  RBAC ClusterRoles so
* authenticated user can create/modify/delete SecretProviderClasses in their namespaces (~ something like reference to a Secret in the 3rd party secret store)
* authenticated user can read SecretProviderClassPodStatuses (~ extension of Pod.Status)